pkg/recipes/engine/engine.go

```diff
@@ -119,14 +128,18 @@ func (e *engine) Delete(ctx context.Context, opts DeleteOptions) error {
 // deleteCore function is the core logic of the Delete function.
 // Any changes to the core logic of the Delete function should be made here.
 func (e *engine) deleteCore(ctx context.Context, recipe recipes.ResourceMetadata, outputResources []rpv1.OutputResource) (*recipes.EnvironmentDefinition, error) {
-	definition, driver, err := e.getDriver(ctx, recipe)
+	configuration, err := e.options.ConfigurationLoader.LoadConfiguration(ctx, recipe)
 	if err != nil {
 		return nil, err
 	}
 
-	configuration, err := e.options.ConfigurationLoader.LoadConfiguration(ctx, recipe)
+	if configuration.Simulated {
```

Comment:
A log here will be helpful as well for debugging, similar to what you have above in execute.